### PR TITLE
[MDB IGNORE] The Grand Airlock Naming Audit

### PR DESCRIPTION
**Pull request by Son-of-Space**
## About The Pull Request

I manually audited all 4,710 instances of airlocks across all maps for upper casing

You'll never guess what I found.

UPDATE: We now have a grep to check for mistitled airlocks, and it's in this PR!

## Why It's Good For The Game

Proper naming convention standards for our players. Good checks prevent inconsistencies in the future.

## Changelog

:cl: Son-of-Space, MrDoomBringer
fix: All airlocks across all maps have been properly titled in upper case
code: A check has been added to look for incorrectly titled airlocks
/:cl:

### CODE REVIEW

**Comment by MrDoomBringer** [comment]
Could you add a grep for this?

**Comment by san7890** [comment]
> Is it possible to add a grep for this?

It is possible, but I'm not sure if it's needed since it's not consistently a problem, just something that has ran unchecked (and should probably be evaluated in upcoming PRs).

For example, grepping for "centcomm" (proper spelling is centcom) makes sense because we will always have people who mis-spell it rather than spell/format it the way we want it to. I don't know how worthwhile it will be to have the grep operation check for stuff like "airlock" versus "Airlock", when there could be a legitmate reason for the non-capitalized "airlock".

**Comment by MrDoomBringer** [comment]
> when there could be a legitmate reason for the non-capitalized "airlock".

There are currently no instances of this situation occurring in our codebase from what I can tell. If in the future this becomes an issue, we can revise the grep.  

> no warranty on this actually working

Your regex indeed seems to match on every instance of the word 'airlock' appearing in a path. 

Instead, this catches about ~300 extra cases that @Son-of-Space seems to have missed with their manual review, and doesn't seem to have any false positives. Please feel free to double-check my work:
`grep -rzoP 'machinery/door.*{([^}]|\n)*name = .*("|\s)(?!of|and|to)[a-z].*\n' --include=*.dmm` 
[airlock_grep_output.txt](https://github.com/tgstation/tgstation/files/8764673/airlock_grep_output.txt)

> I'm not sure if it's needed since it's not consistently a problem

If it needs to be manually evaluated on every future PR, then it's consistently a problem. Piling extra work on the maintainers is unacceptable - especially when this work is easy to miss on a manual review, and trivial to otherwise automate.

**Comment by Son-of-Space** [comment]
I didn't audit machinery/door. I audited machinery/door/airlock

File: tools/ci/check_grep.sh [hunk 1]
```diff
@@ -54,6 +54,11 @@ if grep -Pzo '"\w+" = \(\n[^)]*?/obj/machinery/atmospherics/pipe/(?<type>[/\w]*)
     echo "ERROR: found multiple identical pipes on the same tile, please remove them."
     st=1
 fi;
+if grep -rzoP 'machinery/door.*{([^}]|\n)*name = .*("|\s)(?!of|and|to)[a-z].*\n' --include=*.dmm;	then
```
Comment (san7890) [inline tools/ci/check_grep.sh:57]:
```suggestion
if grep -rzoP 'machinery/door.*{([^}]|\n)*name = .*("|\s)(?!of|and|to)[a-z].*\n' _/maps/**/*.dmm;	then
```

Have it mirror the rest of the greps? This probably works but I just want to make doubly extra sure that something somewhere doesn't break. I know you said you didn't have Windows Subsystem for Linux the other day so if you think it's fine commit this suggestion and I'll test it out on my machine.

Comment (MrDoomBringer) [reply tools/ci/check_grep.sh:57]: The `--include` flag is more portable (doesn't require use of the shopt globstar builtin) and works fine on Ubuntu 20.04. However that's a battle for another day, san is right - better to fit with the other greps for now.

Still, do not commit this, as there's a minor typo - the path should read `_maps/**/*.dmm`

**Comment by san7890** [comment]
image

good it works

**Pull request merged**